include/dpp/discordvoiceclient.h

```diff
@@ -151,11 +154,100 @@ class DPP_EXPORT discord_voice_client : public websocket_client
 	 */
 	std::vector<voice_out_packet> outbuf;
 
-	/**
-	 * @brief Input buffer. Each string is a received UDP
-	 * packet. These will usually be RTP.
-	 */
-	std::vector<std::string> inbuf;
+    /**
+     * @brief Data type of RTP packet sequence number field.
+     */
+	using rtp_seq_num = uint16_t;
+
+    /**
+     * @brief Keeps track of the voice payload to deliver to voice handlers.
+     */
+	struct voice_payload {
+        /**
+         * The sequence number of the RTP packet that generated this voice payload.
+         */
+		rtp_seq_num seq;
+        /**
+         * @brief The event payload that voice handlers receive.
+         */
+		std::unique_ptr<voice_receive_t> vr;
+
+        /**
+         * @brief For priority_queue sorting.
+         */
+		bool operator>(const voice_payload& other) const;
+	};
+
+	struct voice_payload_parking_lot {
+        /**
+         * @brief The minimum RTP packet sequence number we will handle.
+         *
+         * This is to drop packets that arrive too late. Packets before this
+         * number have been delivered to voice handlers and there is no going
+         * back. Unfortunately we just have to drop them.
+         */
+		rtp_seq_num min_seq;
+        /**
+         * @brief Time point when the packet data are flushed and delivered
+         * to clients.
+         *
+         * Each flush causes it to calculate a new next flush time.
+         */
+		std::chrono::steady_clock::time_point next_flush_time;
+        /**
+         * @brief The queue of parked voice payloads.
+         * 
+         * We group payloads and deliver them to handlers periodically as the
+         * handling of out-of-order RTP packets. Payloads in between flushes
+         * are parked and sorted in this queue.
+         */
+		std::priority_queue<voice_payload,
+		                    std::vector<voice_payload>,
+		                    std::greater<voice_payload> // smaller sequence numbers have higher priority
+		                    > parked_payloads;
+#ifdef HAVE_VOICE
+		/**
+		 * @brief libopus decoder
+		 */
+		std::unique_ptr<OpusDecoder,
+		                void(*)(OpusDecoder*)
+		                > decoder{nullptr, nullptr};
+#endif
+	};
+	/**
+	 * @brief Voice buffers to be reported to handler, grouped by speaker.
+	 * Buffers are parked here and flushed when the parking lot hits 500 elemenets.
```

Comment:
Oh this had a typo and the comment is also wrong (old comment). I've changed it to flush every 100ms.